geo/src/algorithm/coords_iter.rs

```diff
@@ -167,6 +226,14 @@ impl<'a, T: CoordinateType + 'a> CoordsIter<'a, T> for Rect<T> {
             y: self.min().y,
         }))
     }
+
+    /// Return the number of coordinates in the `Rect`.
+    ///
+    /// Note: Although a `Rect` is represented by two coordinates, it is
+    /// spatially represented by four, so this method returns `4`.
```

Comment:
Unrelated, but it occurred to me that this is why we have a `map_coords_in_place` method which takes a function rather than a `coors_iter_mut()`. It wouldn't make sense to return 4 mutable coordinates for a Rect. And it would be very surprising if `coords_iter` returned 4 coords while `coords_iter_mut` returned 2 coords.

Reply:
this is a very good point. the change in the pull request (returning `4`) matches the `coords_iter` implementation above. what do you think about us opening a separate issue for this inconsistency? or do you feel like it blocks this pull request?

Reply:
My comment was just referring to an earlier request I'd made long ago that we should add a `coords_iter_mut()`. But now I see that it wouldn't be reasonable to do so.

I think it makes sense that `coords_iter` iterates all the coords from the polygon representation of the rect and I *definitely* think it makes sense that coords_count is consistent with that.

So, I don't see any issue. Certainly nothing blocking this PR - sorry for the confusion.